src/components/FollowButton/index.js

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { connect } from "react-redux";
+import { compose } from "redux";
+import { withHandlers } from "recompose";
+import { withFirestore } from "react-redux-firebase";
+import {
+  following,
+  followRequestSent,
+  followError,
+} from "../../actions/followActions";
+
+const followOnClick = (props) => () => {
+  followHandler(props);
+};
+
+const followHandler = async (props) => {
+  const { firestore, userId, followeeId } = props;
+  const userRef = { collection: "users", doc: userId };
+  const followeeRef = { collection: "users", doc: followeeId };
+  try {
+    const user = await firestore.get(userRef);
+    const followee = await firestore.get(followeeRef);
+    const userData = user.data();
+    const followeeData = followee.data();
+    if (!followeeData.follow.requestOn) {
+      await Promise.all([
+        firestore.update(followeeRef, {
+          "follow.followers": [...followeeData.follow.followers, userId],
+        }),
+        firestore.update(userRef, {
+          "follow.followees": [...userData.follow.followees, followeeId],
+        }),
+      ]);
+      props.following(followeeId);
+    } else {
+      await Promise.all([
+        firestore.update(followeeRef, {
+          "follow.followRequests": [
+            ...followeeData.follow.followRequests,
+            userId,
+          ],
+        }),
+        firestore.update(userRef, {
+          "follow.sentRequests": [...userData.follow.sentRequests, followeeId],
+        }),
```

Comment:
@knguyend You should use the firestore array method here, I think; see [these docs](https://firebase.google.com/docs/firestore/manage-data/add-data#update_elements_in_an_array).

Reply:
Thanks for the doc! Should be good to go!